app/views/devise/registrations/_form.html.haml

```diff
@@ -2,9 +2,10 @@
   =f.error_notification
   =f.input :name
   =f.input :email
-  -unless omniauth_authenticated_and_waiting?
-    =f.input :password
-    =f.input :password_confirmation
+  - if has_email_password?
```

Comment:
Combine for maximum succinctnicity!

`if has_email_password? && !omniauth_authenicated_and_waiting?` :smile: